enclave/cert.c

```diff
@@ -327,19 +321,51 @@ static mbedtls_x509_crt* _sort_certs_by_issue_date(mbedtls_x509_crt* chain)
     return sorted;
 }
 
+/* Call mbedlts_x509_crt_verify and handle error logging */
+static oe_result_t _mbedtls_x509_crt_verify(
+    mbedtls_x509_crt* leaf_cert,
+    mbedtls_x509_crt* ca_cert_chain,
+    mbedtls_x509_crl* ca_crls)
+{
+    oe_result_t result = OE_UNEXPECTED;
+    uint32_t flags = 0;
+    if (mbedtls_x509_crt_verify(
+            leaf_cert, ca_cert_chain, ca_crls, NULL, &flags, NULL, NULL) != 0)
+    {
+        char error[1024] = {0};
+        mbedtls_x509_crt_verify_info(error, sizeof(error), "", flags);
```

Comment:
Check the return value of **mbedtls_x509_crt_verify()**?

Reply:
This call is for mbedtls_x509_crt_verify_**info** which returns the length of the written error string (which is null terminated regardless of success). Just maintaining the existing logic of printing the OE error message with whatever error info can be salvaged from the message, even if that is the empty string. (This is also how mbedtls internally invokes this function)

Reply:
Yeah, besides, this funciton probably cannot fail.